jupyter_client/tests/test_multikernelmanager.py

```diff
@@ -230,6 +230,39 @@ async def test_shutdown_all(self):
         # shutdown again is okay, because we have no kernels
         await km.shutdown_all()
 
+    @gen_test(timeout=20)
+    async def test_use_after_shutdown_all(self):
+        km = self._get_tcp_km()
+        kid = await km.start_kernel(stdout=PIPE, stderr=PIPE)
+        self.assertIn(kid, km)
+        await km.shutdown_all()
+        self.assertNotIn(kid, km)
+
+        # Start another kernel
+        kid = await km.start_kernel(stdout=PIPE, stderr=PIPE)
+        self.assertIn(kid, km)
+        await km.shutdown_all()
+        self.assertNotIn(kid, km)
+        # shutdown again is okay, because we have no kernels
+        await km.shutdown_all()
+
+    @gen_test(timeout=20)
+    async def test_shutdown_all_while_starting(self):
+        km = self._get_tcp_km()
+        kid_future = km.start_kernel(stdout=PIPE, stderr=PIPE)
```

Comment:
I believe we should wrap this result in an `ensure_future()`.  After doing so, I was able to run the tests on 3.6 successfully multiple times w/o failures.
```suggestion
        kid_future = asyncio.ensure_future(km.start_kernel(stdout=PIPE, stderr=PIPE))
```